src/server/api/routes/nft/mint-subscription.js

```diff
@@ -70,10 +78,31 @@ router.get('/nft/mint-subscription', async (req, res, next) => {
 });
 
 router.delete('/nft/mint-subscription/:id', async (req, res, next) => {
-  const { id: subscriptionId } = req.params;
   try {
-    await nftMintSubscriptionCollection.doc(subscriptionId).delete();
-    res.sendStatus(200);
+    const { email } = req.query;
+    if (!email) {
+      res.status(400).send('MISSING_SUBSCRIBER_EMAIL');
+      return;
+    }
+
+    const { id: subscriptionId } = req.params;
+    const docRef = nftMintSubscriptionCollection.doc(subscriptionId);
+    await db.runTransaction(async t => {
+      const doc = await t.get(docRef);
+      if (!doc.exists) {
+        res.status(404).send('SUBSCRIPTION_NOT_FOUND');
+        return;
+      }
+
+      const { subscriberEmail } = doc.data();
+      if (email !== subscriberEmail) {
+        res.status(404).send('SUBSCRIPTION_NOT_FOUND');
+        return;
+      }
+
+      await t.delete(docRef);
```

Comment:
No need for transaction in `delete()` if it works anyway with `try catch` without `get()`